lazy-pages/src/lib.rs

```diff
@@ -124,19 +124,33 @@ pub enum InitializeForProgramError {
     Mprotect(MprotectError),
     #[display(fmt = "Wasm memory end addr is out of usize: begin addr = {_0:#x}, size = {_1:#x}")]
     WasmMemoryEndAddrOverflow(usize, u32),
+    #[display(fmt = "Prefix of storage with memory pages was not set")]
+    MemoryPagesPrefixNotSet,
 }
 
 pub fn initialize_for_program(
     wasm_mem_addr: Option<usize>,
     wasm_mem_size: WasmPageNumber,
     stack_end_page: Option<WasmPageNumber>,
-    program_prefix: Vec<u8>,
+    program_id: [u8; 32],
 ) -> Result<(), InitializeForProgramError> {
     use InitializeForProgramError::*;
     LAZY_PAGES_CONTEXT.with(|ctx| {
         let mut ctx = ctx.borrow_mut();
+
+        let prefix = match ctx.program_storage_prefix.take() {
```

Comment:
Please add a comment than here, that nobody can move this after
```rust
*ctx = LazyPagesExecutionContext::default();
```

Reply:
Also can be problem: in `init_for_program` we take `program_storage_prefix` and add `program_id`. If `init` is called for each message handling then this will not cause the problems, but if we will call `init` once per process queue cycle, then `program_id`s will be accumulated in `program_storage_prefix`, this is very dangerous bug, because it can lead to node binary panic.

Reply:
Hmm, my previous comments disappear...

We have accumulating problem here. If we will make `init` once per process queue, then `program_storage_prefix ` will accumulate `program_id`'s for each handled message.